docs/guides/code-style-guide.md

```diff
@@ -0,0 +1,223 @@
+
+Do the variable names help explain what the code is supposed to do?
+:::
+
+### Use `camelCase`
+
+When writing JavaScript, programmers tend to use a convention called _`camelCase`_. This means starting your variable name with a lower case letter, and then every "word" after that starts with an upper case letter.
+
+- `startingLocation`
+- `timeToDestination`
+- `trafficOnRoute`
+
+By doing this it makes it easier to read variable name that have multiple "words".
+
+### Avoid short names
+
+Very short variable names can be difficult to understand since the purpose of the variable can be unclear. They are also difficult remember, especially if there are many similarly named variables. Try to avoid short names or abbreviations.
+
+Here are some examples of **bad** names that you should avoid:
+
+- Single letters like `x` or `y`
+- Abbreviations like `evt` instead of `event`
+- Generic names like `array` or `string`
+
+### Describe what the variable is/does
+
+A good variable name quickly explains what it represents to anyone reading the code. Try to describe what the variable is or what it does within in code.
+
+```js
+// 🛑 Try to avoid this
+let song = true;
+
+// ✅ This is better
+let isPlaying = true;
+```
+
+In this example, the variable name `song` doesn't tell us what it is used for. The `isPlaying` variable name is better since it tells us whether a song is playing or not.
+
+```js
+// 🛑 Try to avoid this
+function percentage() {
+  // ...
+}
+
+// ✅ This is better
+function getPercentage() {
+  // ...
+}
+```
+
+In this example, the function calculates a percentage and returns it. Therefore it is good to name it using the "get" verb to show that it returns something.
+
+```js
+// 🛑 Try to avoid this
+function isOldEnough(number) {
+  // ...
+}
+
+// ✅ This is better
+function isOldEnough(yearOfBirth) {
+  // ...
+}
+```
+
+Finally, parameters of functions should also have names that properly represent what is going to be received into the function.
+
+## Advanced
+
+These additional rules may be harder for you to understand at the start of your programming journey however they are important concepts as your develop.
+
+### Remember your audience
+
+Remember that you are communicating with another programmer!
+
+Don't forget that the person reading your code probably has some understanding of the _context_. If your function is named `getCustomer`, it's probably fine to name variables `name`, `age`, etc instead of `customerName`, `customerAge`. You don't need to explain _everything_.
+
+**Naming variables is more of an art than a science**. Often you need some experience reading other people's code before you get really good at it.
+
+:::tip
+When reading other people's code, think about the variable names. If the names are not clear, make sure you understand the code first, then think about what names you would use instead.
+:::
+
+### Good scoping of your variables
```

Comment:
@illicitonion @40thieves 

Added your idea here, I think it has value but is a bit more subtle to understand.

I found it hard to describe in simple language _why_ this is important beyond the example given. Happy to take suggestions.

Reply:
Also - I learnt about line highlighting here! - a nice little markdown touch

Reply:
Perhaps it is worth adding something along the lines of (adding on to the last paragraph):

It also means that variables can't be used in confusing places. For example, using `firstName` outside of the `for` loop wouldn't make sense. By keeping the variable inside the loop, it can't be used outside.

I feel like there's some better phrasing to be had, but that's the core idea.